dspace-api/src/main/java/org/dspace/eperson/GroupServiceImpl.java

```diff
@@ -381,6 +395,21 @@ public List<EPerson> allMembers(Context c, Group g) throws SQLException {
         return new ArrayList<>(childGroupChildren);
     }
 
+    @Override
+    public int countAllMembers(Context context, Group group) throws SQLException {
+        // Get all groups which are a member of this group
+        List<Group2GroupCache> group2GroupCaches = group2GroupCacheDAO.findByParent(context, group);
+        Set<Group> groups = new HashSet<>();
```

Comment:
I'm not sure I agree.  With HashSets, the size will automatically double when it reaches 3/4 capacity.  So, if I set the capacity to the full list size, my understanding is it will double in size _before_ I finish adding the list (making it much larger than I need).  I think that can negatively impact our performance here.

See discussion here https://stackoverflow.com/a/38006602

Overall, I've always found/learned that presetting a HashSet's size doesn't help much.  But, maybe you've seen something that I've overlooked?

Reply:
Yeah, I missed the loadfactor in the count, it should be: `new HashSet<>((group2GroupCaches.size()/0.75) + 1);`
Generally you are right about `HashMaps`, but since in this case we are goingo to only populate it using a fixed number of elements, it will be better to declare optimal size at the start.
I think that is high probably to have more than 12 elements (default size is 16, with loadfactor of 0.75 (3/4)), so at least it will be recreated one time, this will decrease performance every time it will be recreated (by doubling its size).
So if we use an optimal init size, since we are using it only to store and query elements, the map won't be resized during the insertions, and all the read operations will be optimal.
Some reference:
- https://stackoverflow.com/a/21822273

Reply:
Good point. I've corrected this (and another instance of HashSet in the same class) in this latest commit: https://github.com/DSpace/DSpace/pull/9078/commits/09e7f105e063ffa7fa35abe7cd9160c59b0f91a8